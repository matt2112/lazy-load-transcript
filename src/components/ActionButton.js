// @flow
import * as React from 'react';
import styled from 'styled-components';

const Button = styled.button`
  height: ${props => (props.size === 'small' ? '40px' : '60px')};
  width: ${props => (props.size === 'small' ? '40%' : '100%')};
  margin-bottom: ${props => (props.size === 'small' ? 0 : '10px')};
  font-size: ${props => (props.size === 'small' ? '1em' : '1.5em')};
  color: ${props => (props.active ? 'white' : 'black')};
  background-color: ${props => (props.active ? '#333' : '#eee')};
  cursor: pointer;
  @media (min-width: 720px) {
    height: ${props => (props.size === 'small' ? '50px' : '70px')};
    width: ${props => (props.size === 'small' ? '20%' : '30%')};
    margin-bottom: 0;
  }
`;

type Props = {
  children: React.Node,
  size: string,
  onClick?: () => void,
  active: boolean
};

const ActionButton = (props: Props) => (
  <Button onClick={props.onClick} {...props}>
    {props.children}
  </Button>
);

ActionButton.defaultProps = {
  onClick: val => val
};

export default ActionButton;
