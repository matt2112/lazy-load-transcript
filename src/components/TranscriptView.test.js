import React from 'react';
import { shallow } from 'enzyme';

import ActionButton from './ActionButton';

it('should render without exploding', () => {
  const wrapper = shallow(<ActionButton />);
  expect(wrapper).toMatchSnapshot();
});

it('should render some text when words stored in state and passed in as props', () => {});
