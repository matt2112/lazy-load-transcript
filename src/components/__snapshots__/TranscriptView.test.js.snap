// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionButton
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
